impexp-core/src/main/java/org/citydb/database/adapter/AbstractUtilAdapter.java

```diff
@@ -184,6 +188,48 @@ public BoundingBox calcBoundingBox(Workspace workspace, List<Integer> objectClas
         }
     }
 
+    public BoundingBox calcBoundingBox(Workspace workspace, Query query, SchemaMapping schemaMapping) throws QueryBuildException, SQLException, FilterException {
+        BoundingBox bbox = null;
+
+        try (Connection conn = databaseAdapter.connectionPool.getConnection()) {
+            if (databaseAdapter.hasVersioningSupport())
+                databaseAdapter.getWorkspaceManager().gotoWorkspace(conn, workspace);
+
+            SQLQueryBuilder builder = new SQLQueryBuilder(
+                    schemaMapping,
+                    databaseAdapter,
+                    BuildProperties.defaults().addProjectionColumn(MappingConstants.ENVELOPE));
+            Select select = builder.buildQuery(query);
+            Table table = new Table(select);
+            select = new Select().addProjection(databaseAdapter.getSQLAdapter().getAggregateExtentFunction(table.getColumn(MappingConstants.ENVELOPE)));
+
+            try (PreparedStatement stmt = databaseAdapter.getSQLAdapter().prepareStatement(select, conn);
+                 ResultSet rs = stmt.executeQuery()) {
+                if (rs.next()) {
+                    Object extentObj = rs.getObject(1);
+                    if (!rs.wasNull() && extentObj != null) {
+                        GeometryObject extent = databaseAdapter.getGeometryConverter().getEnvelope(extentObj);
+                        double[] coordinates = extent.getCoordinates(0);
+                        bbox = new BoundingBox(
+                            new Position(coordinates[0], coordinates[1]),
+                            new Position(coordinates[2], coordinates[3])
+                        );
+                        bbox.setSrs(extent.getSrid());
+
+                        DatabaseSrs targetSrs = query.getTargetSrs();
+                        if (targetSrs != null && bbox.getSrs().getSrid() != targetSrs.getSrid()) {
+                            Tiling tiling = new Tiling(bbox, 1, 1);
+                            tiling.transformExtent(targetSrs, databaseAdapter);
+                            bbox = tiling.getExtent();
```

Comment:
Creating a `Tiling` object just to transform the bbox looks like unecessary overhead. Shouldn't we rather extract just those lines dealing with the transformation from the `Tiling.transformExtent` method and put them in a new method here in the util adapter? Then both this method and  `Tiling.transformExtent` could reuse it. Maybe this new method could even replace the depracted `transformBoundingBox` method in the util adapter?

Reply:
Yes, agree. Maybe we should make this small refactorying in a separate dedicated PR?

Reply:
Agreed